spec/layers.md

```diff
@@ -0,0 +1,99 @@
+clear function
+==============
+**Status**: explainer.
+
+Provide a `beginLayer()` and `endLayer()` functions that create and close layers for Canvas.
+
+
+Rationale
+---------
+
+Currently, the only way to draw a series of drawing calls at the same time, while applying the same sort of effects (shadows, blur, alpha, filters, compositing) to all of the drawn objects, without having these effects being applied one to one to this object, is by using an auxiliary canvas and then, drawing that auxiliary canvas to the canvas.
+This proposal adds an alternative to that (the same way that Fullter in Google and Core Graphics at Apple do) by creating a layer that then will be drawn in one go when the layer ends.
+
+
+Proposal
+--------
+
+```webidl
+interface CanvasRenderingContext2D {
+  // extending original
+  void beginLayer();
+  void endLayer();
+};
+
+interface OffscreenCanvasRenderingContext2D {
+  // extending original
+  void beginLayer();
+  void endLayer();
+};
+```
+The rendering of the layer will use the current state of the canvas, and will reset these effects to the defaults at the end of the layer. The effects that will be used are:
```

Comment:
That sounds impractical.

If these values are really reset to their default at `endLayer()`, how come we can have nested layers? How do you do gCO=source-in inside a gCO=lighter inside a globalAlpha=0.2?  Aren't the gCO=lighter and globalAlpha=0.2 reset by the first inner layer?
Not sure yet what would be the best ergonomics, but should we just take `endLayer()` to mean "render this layer as would drawImage do", i.e using all the context's properties at the time of the call to `endLayer()`? On this case maybe we could bikeshed over `endLayer()` vs `paintLayer()` & al.

Reply:
I think I get what doesn't work here, you probably meant to say that `endLayer()` will reset the properties to the ones saved by the latest `beginLayer()` and not "to the defaults" (here and below)

Reply:
Hi Kaiido, I'm sorry, when you reviewed this explainer, it was still in WIP, and I had some things wrong. Could you take a new look now? Thanks!